src/hooks/useClipboardCopy.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react';
+
+export function useClipboardCopy(timeout = 3000) {
+  const [copied, setCopied] = useState(false);
+
+  const copy = (value: string) => {
+    navigator.clipboard.writeText(value);
+    setCopied(true);
+  };
+
+  useEffect((): (() => void) => {
```

Comment:
It feels like the responsibility of this should be on the Component, not on the hook. As this effect creates an "animation". This should be within the Component